src/flows/components/panel/Results.tsx

```diff
@@ -31,6 +36,19 @@ const Results: FC = () => {
     emptyText = 'No Data Returned'
   }
 
+  const downloadTitle = queryText
+    ? 'Download results as an annotated CSV file'
+    : 'Build a query to download your results'
+  const download = () => {
+    basic(queryText).promise.then(response => {
```

Comment:
i feel hoodwinked. This seems like the perfect opportunity to run the query using the `queryCache` and simply retrieve the query results from there?

Reply:
i'm working on hooking up the query cache work over on the QueryProvider side, still figuring out where variables live because they dont currently exist in notebooks (spoiler: we shouldn't have to pass the whole variable graph just to run a query)

but more to the point here: the reason the query is rerun is because we dont run the full query when trying to grab the data that gets displayed in the simple table interface. Because users normally want a quick view of the shape of the data being returned by their query, that query always has a `|> limit(n: 100)` attached to it. If the user decides that they want to download the results of that query, the idea is that they are then communicating that they want ALL the things, not just the shape, so we want to run it without the limit. Adding the cache in here wouldn't help save any resources.

Reply:
Aw, k. Didn't realize we were limiting the results that get displayed. Guess that's an entirely different issue to overcome but 🤷🏽